samcli/commands/build/build_context.py

```diff
@@ -538,3 +545,17 @@ def _check_java_warning(self) -> None:
 
         if is_building_java and not is_experimental_enabled(ExperimentalFlag.JavaMavenBuildScope):
             click.secho(self._JAVA_BUILD_WARNING_MESSAGE, fg="yellow")
+
+    def _check_esbuild_warning(self) -> None:
+        """
+        Prints warning message and confirms that the user wants to enable beta features
+        """
+        resources_to_build = self.get_resources_to_build()
+        is_building_esbuild = False
+        for function in resources_to_build.functions:
+            if function.metadata and function.metadata.get("BuildMethod", "") == "esbuild":
+                is_building_esbuild = True
+                break
+
+        if is_building_esbuild and not is_experimental_enabled(ExperimentalFlag.Esbuild):
+            prompt_experimental(ExperimentalFlag.Esbuild, self._ESBUILD_WARNING_MESSAGE)
```

Comment:
What if customer denies the prompt? Should we fail and exit?

Reply:
It will fail and exit downstream since the feature flag won't be passed down to lambda builders.
```
Build Failed
Error: Esbuild Failed: Feature flag must be enabled to use this workflow
```